agent/agent.go

```diff
@@ -602,6 +602,9 @@ func (a *Agent) runTask(ctx context.Context, tcInput *taskContext, nt *apimodels
 	defer span.End()
 	tc.traceID = span.SpanContext().TraceID().String()
 
+	tc.dataDisk, err = dataPartition(ctx)
+	grip.Error(errors.Wrapf(err, "getting device name for '%s' partition", dataMountpoint))
```

Comment:
Since this step comes after the task logger is already set up, it would be preferable to send this log to the task logs, since those are things we can easily inspect for debugging if it fails to get the partitions.

Reply:
sent the error to the agent logs